app/routes/_landing.($lang).docs/route.tsx

```diff
@@ -51,37 +59,79 @@ const organizeData = (docs: documentation[]): LinksGroupProps[] => {
       link: doc.slug || "",
       slug: doc.slug || "",
       links: findChildren(doc.id, docs),
+      hasParent: false,
     }))
 }
 
+function flattenTree(tree: LinksGroupProps[]) {
+  let result: LinksGroupProps[] = []
+
+  function dfs(node: LinksGroupProps) {
+    result.push(node)
+    for (let child of node.links) {
+      dfs(child)
+    }
+  }
+
+  for (let node of tree) {
+    dfs(node)
+  }
+
+  return result
+}
+
+function nextNodeInTree(nodes: LinksGroupProps[], node: LinksGroupProps | undefined) {
+  if (!node) return
+
+  const next = nodes.findIndex((n) => n.id === node.id)
+  if (next >= nodes.length - 1) return
+
+  return nodes[next + 1]
+}
+
 export default function DocsLayout() {
   const { data }: LoaderData = useLoaderData()
   const [linksGroupItems, setLinksGroupItems] = useState<LinksGroupProps[]>([])
   const organizeDataRef = useRef(organizeData)
+  const location = useLocation()
+
+  const flattenedTree = useMemo(() => flattenTree(linksGroupItems), [linksGroupItems])
```

Comment:
Final thoughts about the `flattenedTree`... why do we need to flatten it?
`linksGroupItems` was originally generated using `organizedData` function which builds our tree, can't we utilize the data that was originally returned from the server instead of building the tree and then flattening it again?